### PR TITLE
RESI-23 Added Album class, Added AlbumAdapter class, Added row_item f…

### PR DESCRIPTION
**Pull request by leon914**
…or album info holder

### CODE REVIEW

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+
+public class AlbumAdapter extends RecyclerView.Adapter<AlbumAdapter.AlbumViewHolder> {
```
Comment (ScottCooper92) [inline app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: See if you can remove AlbumAdapter. from <AlbumAdapter.AlbumViewHolder> , if not then don't worry about it

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+    private ArrayList<Album> albumList = new ArrayList<>();
```
Comment (ScottCooper92) [inline app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: private List<Album> albumList = new ArrayList();

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    public void setListContent(ArrayList<Album> albumList) {
+        this.albumList = albumList;
+        notifyItemRangeChanged(0, albumList.size());
+    }
+
+    @Override
+    public AlbumViewHolder onCreateViewHolder(final ViewGroup parent, final int viewType) {
+        view = inflater.inflate(R.layout.row_item, parent, false);
```
Comment (ScottCooper92) [inline app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: return LayoutInflater.from(parent.getContext()).inflate(R.layout.row_item, parent, false);

Comment (ScottCooper92) [reply app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: Now you can remove Context from the constructor and remove the LayoutInflator

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import com.squareup.picasso.Picasso;
+
+import java.util.ArrayList;
+
+public class AlbumAdapter extends RecyclerView.Adapter<AlbumAdapter.AlbumViewHolder> {
+
+    private ArrayList<Album> albumList = new ArrayList<>();
+    private final LayoutInflater inflater;
+    View view;
+    AlbumViewHolder holder;
```
Comment (ScottCooper92) [inline app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: Don't store a reference to the holder

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+
+
+
+
+
+
+
+
+//
```
Comment (ScottCooper92) [inline app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: Remove all this commented code

File: app/src/main/java/com/example/lhi06/residue/FindAlbumActivity.java [hunk 6]
```diff
@@ -2,12 +2,31 @@
...
+
+import java.util.ArrayList;
+
+import butterknife.BindView;
 
 public class FindAlbumActivity extends AppCompatActivity {
 
+    AlbumAdapter adapter;
+
+    private ArrayList<Album> listAlbum = new ArrayList<>();
```
Comment (ScottCooper92) [inline app/src/main/java/com/example/lhi06/residue/FindAlbumActivity.java]: List<Album>

File: app/src/main/res/drawable/my_custom.xml [hunk 7]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ScottCooper92) [inline app/src/main/res/drawable/my_custom.xml]: Give this file a proper name

File: app/src/main/res/layout/row_item.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="@dimen/album_art_size"
+    android:orientation="horizontal"
+    android:weightSum="1"
```
Comment (ScottCooper92) [inline app/src/main/res/layout/row_item.xml:6]: What is this for?

File: app/src/main/res/values/dimens.xml [hunk 9]
```diff
@@ -4,4 +4,8 @@
     <dimen name="button_layout_height">60dp</dimen>
     <dimen name="float_button_width">80dp</dimen>
     <dimen name="search_button_margin">0dp</dimen>
+    <dimen name="list_font_size">18dp</dimen>
```
Comment (ScottCooper92) [inline app/src/main/res/values/dimens.xml]: Use sp for text sizes

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java (hunk 1 above)
Comment (leon914) [reply app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: what do you mean sorry? like remove the name of the class?

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java (hunk 2 above)
Comment (leon914) [reply app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: what is the difference just so i know?

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java (hunk 4 above)
Comment (leon914) [reply app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: should i be creating a seperate holder for each method?

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java (hunk 3 above)
Comment (leon914) [reply app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: im not sure what that means but okay 👍

Comment (leon914) [reply app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: ive also done this and its all gone red

File: app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java (hunk 2 above)
Comment (ScottCooper92) [reply app/src/main/java/com/example/lhi06/residue/AlbumAdapter.java]: The benefit being that List is the generic type and can be used to represent different types of lists, from ArrayList to ImmutableList etc. This way you can pass it into functions that modify the list without having to know which specific type it is. If you need me to clarify just ask

File: app/src/main/res/layout/row_item.xml (hunk 8 above)
Comment (leon914) [reply app/src/main/res/layout/row_item.xml:6]: to make sure it stays in the right shape

**Pull request merged**